### PR TITLE
Add ls option to cli

### PR DESCRIPTION
**Pull request by pythops**

### CODE REVIEW

**Comment by tony** [comment]
Nice @pythops ! (Also nice username!)

Can you add a test? (e.g. https://github.com/tmux-python/tmuxp/blob/master/tests/test_cli.py)?

**Comment by pythops** [comment]
Thanks @tony :)
Always a pleasure to contribute.

**Comment by tony** [comment]
@pythops I will take a look at this tomorrow

I need to take a look at why Travis isn't triggering.

**Comment by tony** [comment]
@pythops https://travis-ci.org/github/tmux-python/tmuxp

Can you see this? https://travis-ci.org/github/tmux-python/tmuxp/jobs/702700298

**Comment by tony** [comment]
I will take a peek tomorrow.  It looks like there's another error happening at the same time not related to this PR.

File: tests/test_cli.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 import json
 import os
+from pathlib import Path
```
Comment (tony) [inline tests/test_cli.py:7]: This is an unused import

File: tmuxp/cli.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import logging
 import os
 import sys
+from pathlib import Path
```
Comment (tony) [inline tmuxp/cli.py:13]: Will this work on python 2.7? 

I think we'll do one final release for Python 2.7. If there is a way to do this without `pathlib` that'd be worth checking

File: tmuxp/cli.py [hunk 3]
```diff
@@ -924,3 +925,11 @@ def command_convert(config):
                 buf.write(newconfig)
                 buf.close()
                 print('New config saved to <%s>.' % newfile)
+
+
+@cli.command(name='ls', short_help='List configured sessions in $HOME/.tmuxp dir.')
+def command_ls():
+    tmuxp_dir = Path.home() / '.tmuxp'
```
Comment (tony) [inline tmuxp/cli.py:932]: We have `get_config_dir` to get this path.
```suggestion
    tmuxp_dir = get_config_dir()
```

It respects the user's [`XDG_CONFIG_HOME`](https://wiki.archlinux.org/index.php/XDG_Base_Directory#User_directories) and/or `TMUXP_CONFIGDIR` if its set

File: tmuxp/cli.py [hunk 4]
```diff
@@ -924,3 +925,11 @@ def command_convert(config):
...
+    if tmuxp_dir.exists() and tmuxp_dir.is_dir():
+        for f in tmuxp_dir.iterdir():
```
Comment (tony) [inline tmuxp/cli.py:934]: This is my `.tmuxp` dir: https://github.com/tony/tmuxp-config

When I tried `tmuxp ls`:

```
js-frontend
cpython
LICENSE
eduflow
scripts
braintree
README
_bootstrap_util
python
notifly
peergrade
saltstack
devops
pyterm
freebsd
.gitignore
.git
```

We may want to filter out files that aren't valid extensions (`['.yaml', '.yml', '.json']`, see [here](https://github.com/tmux-python/tmuxp/blob/7649b245fbfcc93ab447a074a4d24f4522493c6b/tmuxp/cli.py#L330))

**Comment by pythops** [comment]
@tony Would you mind restart the tests, they should pass.
Thanks

**Comment by tony** [comment]
@pythops Restarted

**Comment by tony** [comment]
@pythops You can also rebase again - I did several fixes to travis since you last pushed

**Comment by pythops** [comment]
The CI failed because of poetry. Maybe it should be restarted again as I did the rebase after.

**Comment by tony** [comment]
You need to rebase the PR and force push it. It's not easy to give instructions on since I don't know how you have `git remote` names configured.

Can you give me access to your forked repo?  I can do it for you

**Comment by tony** [comment]
@pythops Hm, try this:

`git remote add pythops https://github.com/tmux-python/tmuxp.git`
`git fetch pythops`
`git checkout -b add-ls-option`
`git reset --hard pythops/master`
`git push -u pythops add-ls-option`

This assures that:

1. you are using `pythops` as a remote for your own repo, so we're referring to the same git repo.
2. you aren't using `master` as a branch for a change, since those are better kept in their own specific branches, e.g. `add-ls-option`
3. the branch `add-ls-option` is set-upstream to push to your repo at `add-ls-option` in future commits

You should see an option to open a new pull request, go ahead and do it if you'd like

Now, to rebase so your work is based on the latest tmuxp:

`git remote add tmux-python https://github.com/tmux-python/tmuxp.git`
`git fetch tmux-python`
`git pull --rebase tmux-python master --autostash`
`git push --force-with-lease`

**Comment by pythops** [comment]
@tony you should have access to my fork

**Comment by tony** [comment]
image

**Comment by tony** [comment]
Sorry about that, not sure why the branch closed

**Comment by tony** [comment]
@pythops In what sense do I have access to your repo? I don't seem to be able to push branches

See #616, I PR'd it, but the problem is - I'd sort of take credit for your work.

I want to make sure you get credit for the Pull request/commits. I think you need to redo it and rebase correctly - it's really up to your system and reading up on rebasing, I wrote more here: https://github.com/tmux-python/tmuxp/pull/599#issuecomment-656224617

Do you want to make your own pull request? If so, I think you need to figure out the branching and make a new PR.  If I do it for you, I end up becoming the creator of the PR and part of the commit messages.

**Comment by tony** [comment]
Also sorry for all the trouble with this PR!

Initially our CI system was broke. And rebasing it kind of tricky to explain without me knowing your configuration!

**Pull request closed without merging**